content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV_bounded[3];  // Will be bound to t1 register... Until t3 (inclusive).
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t4 register, and will take over t5+.
+        RWTexture2D<float4>      m_texUAV_bounded[5];  // Will be bound to u0 register... Until u4 (inclusive).
+    	RWTexture2D<float4>      m_texUAV_unbounded[]; // Will be bound to u5 register, and will take over u6+.
+        Sampler                  m_sampler1_bounded[7]; // Will be bound to s0 register... Until s6 (inclusive).
+        Sampler                  m_sampler2_bounded[3]; // Will be bound to s7 register... Until s9 (inclusive).
+        Sampler                  m_sampler_unbounded[]; // Will be bound to s10 register, and will take over s11+.
+        ConstantBuffer<MyStruct> m_cb_array[2]; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register)... Until     b2 (inclusive).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b3 register.
+        ConstantBuffer<MyStruct> m_cb_unbounded[]; // Will be bound to b4 register, and will take over b5+.
+    };
+```
+Now, let's review what will happen to all the examples above when using '--unique-idx', which is the case for Vulkan & Metal.
+
+### Example 6: Based on Example 1, with '--unique-idx'
+Same code as Example 1, but it is assumed to be compiled with '--unique-idx' (ON), which is required by Vulkan & Metal. Please note how the register index always increments, regardless of the resource type.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t2 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u3 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u4 register.
+        Sampler                  m_sampler1;     // Will be bound to s5 register.
+        Sampler                  m_sampler2;     // Will be bound to s6 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b7 register.
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b8 register.
+    };
+```
+### Example 7: Based on Example 2, with '--unique-idx'. Error Case.
```

Comment:
```suggestion
#### Error case

This example demonstrates how compiling an unbounded array (`Texture2D<float4> m_texSRV_unbounded[]`) with the `--use-idx` option results in a register assignment conflict. 

This example fails to compile because the unbounded array, `m_texSRV_unbounded[]`, takes over a whole range of register indices, regardless of resource type. `m_texSRV_unbounded[]` takes over `tN`, `uN`, `sN`, and `bN` registers, as opposed to the first error case in [Example: Register assignments](#error-case) where it only takes over `tN` registers.
```

- Simplify title. Move the example's explanation out of the title and into its own sentence. 
- Further explain this example. What do you want to teach the reader with this example? Consider my suggestion.

Reply:
Fixed.